### PR TITLE
Remove include=stop from api /routes/{id}

### PR DESCRIPTION
**Pull request by lboyarsky**
Ticket: [Remove include=stop from api /routes/{id}](https://app.asana.com/0/810933294009540/1110659645781132)

### CODE REVIEW

**Comment by paulswartz** [comment]
This doesn't seem like quite the right behavior.

- /routes/?filter[stop]=44&include=stop should work, and appear in the documentation for `/routes/`
- /routes/9?include=stop should not work, and not appear in the documentation for `/routes/{id}`

**Comment by lboyarsky** [comment]
I believe it works as you described. 

> /routes/?filter[stop]=44&include=stop should work, and appear in the documentation for `/routes/`
```bash
Levs-MBP:~ lev$ curl 'http://localhost:4000/routes?stop=44&include=stop' | jsonpp
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1827  100  1827    0     0  46051      0 --:--:-- --:--:-- --:--:-- 46846
{
  "data": [
    {
      "attributes": {
        "color": "FFC72C",
        "description": "Community Bus",
        "direction_destinations": [
          "City Point",
          "McCormack Housing"
        ],
        "direction_names": [
          "Outbound",
          "Inbound"
        ],
        "fare_class": "Local Bus",
        "long_name": "City Point - McCormack Housing",
        "short_name": "5",
        "sort_order": 50050,
        "text_color": "000000",
        "type": 3
      },
      "id": "5",
      "links": {
        "self": "/routes/5"
      },
      "relationships": {
        "line": {
          "data": {
            "id": "line-510",
            "type": "line"
          }
        },
        "stop": {
          "data": {
            "id": "44",
            "type": "stop"
          }
        }
      },
      "type": "route"
    },
    {
      "attributes": {
        "color": "FFC72C",
        "description": "Local Bus",
        "direction_destinations": [
          "City Point",
          "Copley"
        ],
        "direction_names": [
          "Outbound",
          "Inbound"
        ],
        "fare_class": "Local Bus",
        "long_name": "City Point - Copley via Broadway Station",
        "short_name": "9",
        "sort_order": 50090,
        "text_color": "000000",
        "type": 3
      },
      "id": "9",
      "links": {
        "self": "/routes/9"
      },
      "relationships": {
        "line": {
          "data": {
            "id": "line-9",
            "type": "line"
          }
        },
        "stop": {
          "data": {
            "id": "44",
            "type": "stop"
          }
        }
      },
      "type": "route"
    },
    {
      "attributes": {
        "color": "FFC72C",
        "description": "Local Bus",
        "direction_destinations": [
          "City Point",
          "Copley"
        ],
        "direction_names": [
          "Outbound",
          "Inbound"
        ],
        "fare_class": "Local Bus",
        "long_name": "City Point - Copley via Boston Medical Center",
        "short_name": "10",
        "sort_order": 50100,
        "text_color": "000000",
        "type": 3
      },
      "id": "10",
      "links": {
        "self": "/routes/10"
      },
      "relationships": {
        "line": {
          "data": {
            "id": "line-510",
            "type": "line"
          }
        },
        "stop": {
          "data": {
            "id": "44",
            "type": "stop"
          }
        }
      },
      "type": "route"
    }
  ],
  "included": [
    {
      "attributes": {
        "address": null,
        "description": null,
        "latitude": 42.335548,
        "location_type": 0,
        "longitude": -71.041754,
        "name": "E Broadway @ H St",
        "platform_code": null,
        "platform_name": null,
        "wheelchair_boarding": 1
      },
      "id": "44",
      "links": {
        "self": "/stops/44"
      },
      "relationships": {
        "child_stops": {},
        "facilities": {
          "links": {
            "related": "/facilities/?filter[stop]=44"
          }
        },
        "parent_station": {
          "data": null
        }
      },
      "type": "stop"
    }
  ],
  "jsonapi": {
    "version": "1.0"
  }
}
```
image

>  /routes/9?include=stop should not work, and not appear in the documentation for `/routes/{id}`

```bash
Levs-MBP:~ lev$ curl 'http://localhost:4000/routes/9?include=stop' | jsonpp
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   471  100   471    0     0   8483      0 --:--:-- --:--:-- --:--:--  8563
{
  "data": {
    "attributes": {
      "color": "FFC72C",
      "description": "Local Bus",
      "direction_destinations": [
        "City Point",
        "Copley"
      ],
      "direction_names": [
        "Outbound",
        "Inbound"
      ],
      "fare_class": "Local Bus",
      "long_name": "City Point - Copley via Broadway Station",
      "short_name": "9",
      "sort_order": 50090,
      "text_color": "000000",
      "type": 3
    },
    "id": "9",
    "links": {
      "self": "/routes/9"
    },
    "relationships": {
      "line": {
        "data": {
          "id": "line-9",
          "type": "line"
        }
      },
      "stop": {
        "data": null
      }
    },
    "type": "route"
  },
  "jsonapi": {
    "version": "1.0"
  }
}
```
image

If you mean that it should return `:bad_include`, then I can also do this but in this case it will need to be changed everywhere. What I have implemented in https://github.com/mbta/api/pull/11 is checking for includes only for those calls, which support them. 
```

**Comment by paulswartz** [comment]
There are valid includes for `show`: https://api-v3.mbta.com/routes/441?include=line

File: apps/api_web/lib/api_web/controllers/route_controller.ex [hunk 1]
```diff
@@ -13,7 +13,8 @@ defmodule ApiWeb.RouteController do
 
   @filters ~w(id stop type direction_id date)
   @pagination_opts [:offset, :limit, :order_by]
-  @includes ~w(stop)
+  @includes_show ~w(stop)
+  @includes_index ~w(line) ++ @includes_show
```
Comment (paulswartz) [inline apps/api_web/lib/api_web/controllers/route_controller.ex]: This is the right idea, but backwards: `line` is the only include for `show`, and `stop` is also available for `index`.

**Comment by lboyarsky** [comment]
Ok, it was not in the swagger though. Fixed. 
Probably we should have a tech debt ticket to sync swagger with reality.

File: apps/api_web/lib/api_web/controllers/route_controller.ex (hunk 1 above)
Comment (lboyarsky) [reply apps/api_web/lib/api_web/controllers/route_controller.ex]: Yes, just fixed it

**Comment by paulswartz** [comment]
:+1: filed a ticket for that

**Pull request merged**